tests-behat/vue.feature

```diff
@@ -1,8 +1,11 @@
 Feature: Vue
   Testing Vue component
 
-  Scenario: testing InlineEdit
+  Scenario: loading components
     Given I am on "javascript/vue-component.php"
+    Then I check for vue components to be load
```

Comment:
this is basically a wait we want to avoid in any circumstances as it will be needed before any component without standard JS

we may want to tweak the Context::jqueryWait and/or Context::getFinishedScript methods instead

can we somehow monitor loading JS?

I think we can:
1. set some global/app flag with that JS name
2. unset this flag right at the end? in that JS file
3. in Context::getFinishedScript check if no JS loading flag is set

Reply:
Not sure what you have in mind here. To me, this is what it does.

Vue component js file is now loaded dynamically when required.

This is the cycle of operation:
- createAtkVue method is creating the Vue instance;
- When It does, it set the flag for this component 'isLoaded' to false;
- Then js file need for this component start loading and when the file is fully loaded, 'isLoaded' for that component is set to true;

calling `atk.vueService.areComponentsLoaded()` will return true only when all components file requests are loaded, otherwise false if still trying to load some files.

What this test does is only checking if all file are loaded using:
 `$loaded = $this->getSession()->wait(2000, 'atk.vueService.areComponentsLoaded()');`

It will wait until the javascript expression above return true, or until the max time of 2sec is reached. 
In most of the case, there is no waiting time, cause test pass most of the time. If not, it will probably wait just a few ms or until all files are loaded. Which is needed to run the test.

Reply:
I see `atk.vueService.areComponentsLoaded()` - can we simply add it into `Context::getFinishedScript`?

Reply:
yes, for vue only

Reply:
good, I pushed the change to my biggest concern - I do not want to see any wait in tests - all loading conditions must be checked on the fly, not manually